src/main/java/org/ga4gh/drs/utils/objectloader/FileDrsObjectLoader.java

```diff
@@ -1,21 +1,109 @@
 
 public class FileDrsObjectLoader extends AbstractDrsObjectLoader {
 
-    //TODO populate this class
+    private File file;
+
+    public FileDrsObjectLoader(String objectId, String objectPath) {
+        super(objectId, objectPath);
+        file = new File(objectPath);
+    }
 
     public boolean exists() {
-        return false;
+        return getFile().exists();
     }
 
-    public DrsObject getImputedProperties() {
-        return null;
+    public boolean isBundle() {
+        return getFile().isDirectory();
+    }
+
+    public List<AccessMethod> generateAccessMethods() {
+        // TODO currently access ID has no meaning within the app, should be
+        // used to populate a lookup cache
+        String accessID = UUID.randomUUID().toString();
+        AccessMethod accessMethod = new AccessMethod(accessID, AccessType.HTTPS);
+        return new ArrayList<AccessMethod>() {{add(accessMethod);}};
     }
 
-    public DrsObject getExplicitProperties() {
+    public List<ContentsObject> generateContents() {
+        // TODO fill out stub method
         return null;
     }
-    
+
+    public DrsObject generateCustomDrsObjectProperties() {
+        DrsObject customDrsObject = new DrsObject();
+        String customDrsObjectFilePath = getObjectPath() + ".drsmeta.json";
+        File customDrsObjectFile = new File(customDrsObjectFilePath);
+        if (customDrsObjectFile.exists() && !customDrsObjectFile.isDirectory()) {
+            try {
+                ObjectMapper mapper = new ObjectMapper(new JsonFactory());
+                customDrsObject = mapper.readValue(customDrsObjectFile, DrsObject.class);
+            } catch (IOException e) {
+                // TODO log the exception
+                return customDrsObject;
+            }
+        }
+        return customDrsObject;
+    }
+
+    public List<Checksum> imputeChecksums() {
+        List<Checksum> checksums = new ArrayList<>();
+        try (InputStream is = Files.newInputStream(getFile().toPath())) {
+            String digest = DigestUtils.md5DigestAsHex(is);
+            Checksum md5Checksum = new Checksum(digest, "md5");
+            checksums.add(md5Checksum);
+        } catch (IOException e) {
+            return checksums;
+        }
+        return checksums;
+    }
+
+    public long imputeSize() {
+        return getFile().length();
+    }
+
+    public String imputeName() {
+        return getFile().getName();
+    }
+
+    public String imputeMimeType() {
+        String ext = FilenameUtils.getExtension(getObjectPath());
+        return MimeTypeImputer.imputeMimeType(ext);
+    }
+
+    public LocalDateTime imputeCreatedTime() {
+        LocalDateTime createdTime = null;
+        try {
+            FileTime creationTime = (FileTime) Files.getAttribute(getFile().toPath(), "creationTime");
+            createdTime = LocalDateTime.ofInstant(creationTime.toInstant(), ZoneId.of("UTC"));
+        } catch (IOException e) {
+            return createdTime;
```

Comment:
I wonder if we should use 0 seconds from Unix epoch in the case where we can't get an accurate time because
* the json serialization scheme we use will omit null fields
* the spec requires `created_time` so omitting the field would be a malformed object